File: 프로그래머스/1/340213. ［PCCP 기출문제］ 1번 ／ 동영상 재생기/［PCCP 기출문제］ 1번 ／ 동영상 재생기.py
def timecalculator(time):
    minute, second = map(int, time.split(":"))
    return minute * 60 + second
def solution(video_len, pos, op_start, op_end, commands):
    videotime=timecalculator(video_len)
    postime=timecalculator(pos)
    opstime=timecalculator(op_start)
    opetime=timecalculator(op_end)
    if opstime<=postime<=opetime:
        postime=opetime
    for i in commands:
        if i=="prev":
            if postime<10:
                postime=0
            else:
                postime-=10
        elif i=="next":
            if videotime-postime<10:
                postime=videotime
            else:
                postime+=10
        if opstime<=postime<=opetime:
            postime=opetime
    minute,second=postime//60,postime%60
    return f"{minute:02}:{second:02}"

File: 프로그래머스/1/340213. ［PCCP 기출문제］ 1번 ／ 동영상 재생기/［PCCP 기출문제］ 1번 ／ 동영상 재생기.java
class Solution {
    private int timeConverter(String time){
        String[] splittime = time.split(":");
        return Integer.parseInt(splittime[0])*60+Integer.parseInt(splittime[1]);
    }

    private int skip(int time, int opSec, int opEndSec){
        if(opSec <= time && time<= opEndSec){
            time=opEndSec;
        }
        return time;
    }
    public String solution(String video_len, String pos, String op_start, String op_end, String[] commands) {
        

        int videoSec = timeConverter(video_len);
        int posSec = timeConverter(pos);
        int opSec = timeConverter(op_start);
        int opEndSec = timeConverter(op_end);
        posSec = skip(posSec, opSec, opEndSec);

        for(String command : commands){
            if(command.equals("next"))
            {
                posSec +=10;
                if (posSec > videoSec){
                    posSec = videoSec;
                }
                posSec=skip(posSec, opSec, opEndSec);
            }
            if(command.equals("prev"))
            {
                posSec -= 10;
                if (posSec < 0) {
                    posSec = 0;
                }
                posSec = skip(posSec, opSec, opEndSec);
            }
        }
        String answer = String.format("%02d:%02d", posSec / 60, posSec % 60);
        return answer;
    }
}